File: Patterns/Is Thank You/Example Adapters/Boilerplate_Is_Thank_You_Adapter.py
from flask import Flask, jsonify, request

app = Flask(__name__)

# Route for /TestConfiguration
@app.route('/TestConfiguration', methods=['POST'])
def test_configuration():
    try:
        payload = request.get_json()
        # Perform the configuration test here
        
        # Return a successful response
        return jsonify(True), 200
    except:
        # Return an error response if the test fails or an exception occurs
        return jsonify(message="Test failed"), 500

# Route for /IsThankYou
@app.route('/IsThankYou', methods=['POST'])
def is_thank_you():
    try:
        payload = request.get_json()
        # available payload attributes
        # payload["AdapterId"]
        # payload["ConfigurationId"]
        # payload["Guid"]
        # payload["PacketGUID"]
        # payload["From"]
        # payload["To"]
        # payload["Subject"]
        # payload["Body"]
        # payload["SentOn"]
        # payload["ProcessedOn"]
        # payload["OriginalStatus"]
        # payload["EnateInstance"]

        # Perform the is thank you prediction here
        # Typically this is where the Body and/or Subject are sent to a 3rd
        # party service for classification.
        # The 3rd party service will return a custom/proprietary JSON body which
        # will then need to be converted to schema below.

        # Return the prediction result
        result = {
            "Guid": payload['Guid'],
            "ProcessedOn": payload['ProcessedOn'],
            "PredictionScore": 0.76
        }
        return jsonify(result), 200
    except:
        # Return an error response if the prediction fails or an exception occurs
        return jsonify(message="Prediction failed"), 500

if __name__ == '__main__':
    app.run()

File: Patterns/Is Thank You/Example Adapters/Boilerplate_Is_Thank_You_Adapter.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace YourNamespace
{
    public static class ThankYouApi
    {
        [FunctionName("PostTestConfiguration")]
        public static async Task<IActionResult> TestConfiguration(
            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "TestConfiguration")] HttpRequest req,
            ILogger log)
        {
            try
            {
                // Parse and validate the request body
                var configuration = await req.Content.ReadAsAsync<AdapterConfigurationEntryResult>();

                // Perform the test configuration logic
                // ...

                // Return the response based on the test result
                return new OkObjectResult(true);
            }
            catch (Exception)
            {
                return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
            }
        }

        [FunctionName("PostIsThankYou")]
        public static async Task<IActionResult> IsThankYou(
            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "IsThankYou")] HttpRequest req,
            ILogger log)
        {
            try
            {
                // Parse and validate the request body
                var packetCommunication = await req.Content.ReadAsAsync<PacketCommunicationForIsThankYou>();

                // Perform the is thank you email prediction logic
                // ...

                // Prepare and return the prediction result
                var predictionResult = new PredictionResult
                {
                    Guid = packetCommunication.Guid,
                    ProcessedOn = packetCommunication.ProcessedOn,
                    PredictionScore = 0.76 // Replace with the actual prediction score
                };

                return new OkObjectResult(predictionResult);
            }
            catch (Exception)
            {
                return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
            }
        }
    }

    public class AdapterConfigurationEntryResult
    {
        [JsonProperty("Name")]
        public string Name { get; set; }
        [JsonProperty("Value")]
        public object Value { get; set; }
    }

    public class PacketCommunicationForIsThankYou
    {
        [JsonProperty("AdapterId")]
        public string AdapterId { get; set; }
        [JsonProperty("ConfigurationId")]
        public string ConfigurationId { get; set; }
        [JsonProperty("Guid")]
        public string Guid { get; set; }
        [JsonProperty("PacketGuid")]
        public string PacketGuid { get; set; }
        [JsonProperty("From")]
        public string From { get; set; }
        [JsonProperty("To")]
        public string To { get; set; }
        [JsonProperty("Subject")]
        public string Subject { get; set; }
        [JsonProperty("Body")]
        public string Body { get; set; }
        [JsonProperty("SentOn")]
        public string SentOn { get; set; }
        [JsonProperty("ProcessedOn")]
        public string ProcessedOn { get; set; }
        [JsonProperty("OriginalStatus")]
        public int OriginalStatus { get; set; }
        [JsonProperty("EnateInstance")]
        public EnateIdentityResult EnateInstance { get; set; }
    }

    public class PredictionResult
    {
        [JsonProperty("Guid")]
        public string Guid { get; set; }
        [JsonProperty("ProcessedOn")]
        public string ProcessedOn { get; set; }
        [JsonProperty("PredictionScore")]
        public double PredictionScore { get; set; }
    }

    public class EnateIdentityResult
    {
        [JsonProperty("EnateID")]
        public string EnateID { get; set; }
        [JsonProperty("Name")]
        public string Name { get; set; }
        [JsonProperty("Subject")]
        public string Subject { get; set; }
        [JsonProperty("URL")]
        public string URL { get; set; }
        [JsonProperty("EnateIdentityCertificateName")]
        public string EnateIdentityCertificateName { get; set; }
    }
}